crates/solidity/outputs/cargo/crate/src/generated/kinds.rs

```diff
@@ -18,144 +18,213 @@ use {napi::bindgen_prelude::*, napi_derive::napi};
 #[cfg_attr(not(feature = "slang_napi_interfaces"), derive(Clone, Copy))]
 pub enum ProductionKind {
     ABICoderPragma,
+    AdditiveExpression,
     AddressType,
+    AndExpression,
     ArgumentsDeclaration,
     ArrayExpression,
-    ArrayValuesList,
-    AsciiStringLiteralsList,
-    AssemblyFlagsList,
+    ArrayTypeName,
+    ArrayValues,
+    AsciiStringLiterals,
+    AssemblyFlags,
+    AssemblyFlagsDeclaration,
     AssemblyStatement,
+    AssignmentExpression,
+    BitwiseAndExpression,
+    BitwiseOrExpression,
+    BitwiseXorExpression,
     Block,
     BreakStatement,
     CatchClause,
     CatchClauseError,
-    CatchClausesList,
+    CatchClauses,
+    ComparisonExpression,
+    ConditionalExpression,
     ConstantDefinition,
-    ConstructorAttributesList,
+    ConstructorAttribute,
+    ConstructorAttributes,
     ConstructorDefinition,
     ContinueStatement,
     ContractDefinition,
-    ContractMembersList,
-    DeconstructionImport,
-    DeconstructionImportSymbol,
-    DeconstructionImportSymbolsList,
+    ContractMember,
+    ContractMembers,
+    DecimalNumberExpression,
+    DefaultAllKeywords,
     DeleteStatement,
     DoWhileStatement,
+    ElementaryType,
+    ElseBranch,
     EmitStatement,
-    EndOfFileTrivia,
     EnumDefinition,
+    EnumMembers,
+    EqualityExpression,
     ErrorDefinition,
     ErrorParameter,
-    ErrorParametersList,
+    ErrorParameters,
+    ErrorParametersDeclaration,
     EventDefinition,
     EventParameter,
-    EventParametersList,
+    EventParameters,
+    EventParametersDeclaration,
+    ExperimentalFeature,
     ExperimentalPragma,
+    ExponentiationExpression,
     Expression,
     ExpressionStatement,
-    FallbackFunctionAttributesList,
+    FallbackFunctionAttribute,
```

Comment:
I see quite a few new kinds here. For example, both `ExponentiationExpression` abd `BinaryExpression`. Is it using `PrecedenceExpression.name` or `PrecedenceExpression.rule_name`?

Reply:
These are using `rule_name` for new rule kinds introduced by the precedence operators.

In DSL v1 we had inline parser operators that were bundled under a single, named rule. In DSL v2-to-v1, we have separate each operator as a parser (so dedicated rule) but use the introduced `rule_name` to bridge the gap, since the v1 model expects a grouping rule node.

We could solve this differently by introducing an explicit grouping for the PrecedenceExpressions with a given name, but I don't think we can get rid of it altogether, if we want to use the existing grouping rule kinds like `BinaryExpression` for CST or EBNF.

In general, I didn't re-introduce inlined parsers except for the choices, so that's one of the reasons we have so many new kinds. We should probably think if we can limit the impact here, e.g. by treating the precedence operators as inline parsers.

However, we can't easily retain the old behaviour, as the new v2 model has to now name some previously inlined sequences, leading to new kinds. We could solve this by adding some new annotations like `parser = (inline = true)` for structs and the like, but I'm somewhat on the fence here - I'm worried this will add more noise.

Solving this in a more principled way would require supporting optional struct inlining. At the moment it seems like a lot of work for not that much benefit, as we wouldn't have to deal with transforming nominal types when defining the AST but also tuples.

Reply:
The precedence operators are now inlined again in https://github.com/NomicFoundation/slang/pull/650/commits/8f0cc2d594554fdf9a6e03350580ac53100b9b20, so this only introduces kinds for now-named, previously-inlined sequences (e.g. `UsingAlias`).

Reply:
Thanks for explaining!